parser.go

```diff
@@ -171,22 +177,53 @@ func ParseEventMetadataFromTransform(transform casper.Transform) (EventMetadata,
 
 // FetchContractSchemasBytes accept contract hash to fetch stored contract schema
 func (p *EventParser) FetchContractSchemasBytes(contractHash casper.Hash) ([]byte, error) {
-	schemasURefValue, err := p.casperClient.QueryGlobalStateByStateHash(context.Background(), nil, fmt.Sprintf("hash-%s", contractHash.ToHex()), []string{eventSchemaNamedKey})
+	loadContractSchemasFromEntity := func(contractHash casper.Hash) (rpc.QueryGlobalStateResult, error) {
+		entity, err := p.casperClient.GetLatestEntity(context.Background(), rpc.EntityIdentifier{
+			EntityAddr: &key.EntityAddr{
+				SmartContract: &contractHash,
+			},
+		})
+		if err != nil {
+			return rpc.QueryGlobalStateResult{}, err
+		}
+
+		addressableEntity := entity.Entity.AddressableEntity
+		if addressableEntity != nil && addressableEntity.Entity.EntityKind.SmartContract != nil {
+
+			var eventSchemaUref string
+			for _, namedKey := range addressableEntity.NamedKeys {
+				if namedKey.Name == eventSchemaNamedKey {
+					eventSchemaUref = namedKey.Key.String()
+					break
+				}
+			}
+
+			schemasURefValue, err := p.casperClient.QueryGlobalStateByStateHash(context.Background(), nil, eventSchemaUref, nil)
+			if err != nil {
+				return rpc.QueryGlobalStateResult{}, err
+			}
+
+			return schemasURefValue, nil
+		}
+		return rpc.QueryGlobalStateResult{}, ErrNotSmartContractAddressableEntity
+	}
+
+	schemasURefValue, err := loadContractSchemasFromEntity(contractHash)
 	if err != nil {
-		return nil, err
+		log.Println("Error pn fetching schemas bytes from addressable entity: ", err)
+
+		schemasURefValue, err = p.casperClient.QueryGlobalStateByStateHash(context.Background(), nil, fmt.Sprintf("hash-%s", contractHash.ToHex()), []string{eventSchemaNamedKey})
```

Comment:
@ZhmakAS, it probably would make more sense to provide the possibility to specify the version and use the dedicated approach for 1.5.x and 2.0.0. With fallbacks, we won't be able to use the parser efficiently in 1.5.x